cmd/clusterctl/client/repository/template.go

```diff
@@ -92,9 +93,12 @@ func NewTemplate(input TemplateInput) (*template, error) {
 		}, nil
 	}
 
-	processedYaml, err := input.Processor.Process(input.RawArtifact, input.ConfigVariablesClient.Get)
-	if err != nil {
-		return nil, err
+	processedYaml := input.RawArtifact
+	if !input.SkipVariables {
+		processedYaml, err = input.Processor.Process(input.RawArtifact, input.ConfigVariablesClient.Get)
```

Comment:
Another potential backside of this option is that the RawArtifact might not be something that can be converted directly into a list of Unstructured objects, so the rest of this function could fail ... 🤔

Reply:
I think it makes sense to rename to `skipProcess`. Following my comment above, do you think this should also be renamed for components?

Reply:
> Another potential backside of this option is that the RawArtifact might not be something that can be converted directly into a list of Unstructured objects, so the rest of this function could fail ... 🤔

Should we return early if `skipProcess` is enabled to avoid this?

Reply:
> Should we return early if skipProcess is enabled to avoid this? As in just populate variables and namespace in template, and return?

If possible yes

Reply:
Thinking more about this.
If we are returning at this stage, what is the difference between ListVariablesOnly and SkipProcess? should we merge those behaviors?